### DISCUSSION TITLE
moving channel position

### DISCUSSION THREAD

**Discussion by allomanticpush** [Rapptz/discord.py#6493] [Q&A]
I apologize for my noobish-ness, but I think asking for help will be much better than banging my head against the problem for the extended future.

Is it possible to track channel position moves? Hopefully, My server would like to catch who is rearranging the channels on a daily basis. It has been going on for weeks, and has really worn out the little bit of humor it started with. 

Please let me know where to get started on this, thanks.

AllomanticPush

**Comment by SebbyLaw** [comment]
> My server would like to catch who is rearranging the channels

This is a discord limitation, and not possible.

It's possible to track *when* a channel position changes with [`on_guild_channel_update`](https://discordpy.readthedocs.io/en/latest/api.html#discord.on_guild_channel_update):
```py
@bot.event
async def on_guild_channel_update(before, after):
    if before.position != after.position:
        print(f"channel {after} was moved from {before.position} to {after.position}!")
```
However, discord does not provide enough data to know *who* changed the channel's position and this is not possible.

**Reply by allomanticpush** [reply]
Yeah, sort of what I thought, that it was a limitation from discord first. 

Thank you for the code, we will use it and hopefully can figure out when it’s happening, and then who. 

Thanks again,

allomanticpush
